lib/govspeak/remove_advisory_service.rb

```diff
@@ -0,0 +1,58 @@
+module Govspeak
+  class RemoveAdvisoryService
+    attr_reader :object, :regex, :whodunnit
+
+    def initialize(object, regex, whodunnit = User.find_by(name: "GDS Inside Government Team"))
+      @object = object
+      @regex = regex
+      @whodunnit = whodunnit
+    end
+
+    def process!
+      body = object.body
+      matches = body.scan(regex)
+      return if matches.empty?
+
+      new_body = body.gsub(/(^@)([\s\S]*?)(@?)(?=(?:^\$CTA|\r?\n\r?\n|^@|$))/m) do
+        content = Regexp.last_match(1)
+        "^#{content}^"
+      end
+      if new_body != body
+        record.update!(body: new_body)
+      end
+    end
+
+  private
+
+    def create_minor_update(new_body)
```

Comment:
If I understand it correctly, we don't need this method as we are not going to be handling drafts that contain advisory markup anymore - is that right?

If so you could simplify this class, and just initialise it with an `edition`? 
- You could have a method that returns the regex, rather than passing it in as it's not needed elsewhere, only in this class. 
- You could still have the `@body` variable if you did
```
def initialize(object)
  @body = object.body
end

attr_reader :body
```

Reply:
I think it would be good to keep this in, as it will match our flow for normal editions. So the changes will appear in the audit history for the doc